eval/src/vespa/eval/tensor/dense/dense_single_reduce_function.cpp

```diff
@@ -147,14 +137,99 @@ struct MyGetFun {
 
 using MyTypify = TypifyValue<TypifyCellType,TypifyAggr,TypifyBool>;
 
+std::pair<std::vector<vespalib::string>,ValueType> sort_and_drop_trivial(const std::vector<vespalib::string> &list_in, const ValueType &type_in) {
+    std::vector<vespalib::string> dropped;
+    std::vector<vespalib::string> list_out;
+    for (const auto &dim_name: list_in) {
+        auto dim_idx = type_in.dimension_index(dim_name);
+        assert(dim_idx != ValueType::Dimension::npos);
+        const auto &dim = type_in.dimensions()[dim_idx];
```

Comment:
i think you've mentioned that we could have these three lines as a utility in the type
`type_in.dimension(dim_name)`?